tla/consensus/ccfraft.tla

```diff
@@ -1054,7 +1054,8 @@ HandleAppendEntriesResponse(i, j, m) ==
           /\ leadershipState[i] = Leader \* Only Leaders need to tally append entries responses
           /\ m.success \* successful
           \* max(...) because why would we ever want to go backwards on a success response?!
-          /\ matchIndex' = [matchIndex EXCEPT ![i][j] = max(@, m.lastLogIndex)]
+          \* min(...) because we don't want to calculate a match past our own log's tail
+          /\ matchIndex' = [matchIndex EXCEPT ![i][j] = max(@, min(Len(log[i]), m.lastLogIndex))]
```

Comment:
We've had this in `raft.h` for a while:

```
        // Potentially unnecessary safety check - use min with last_idx, to
        // prevent matches past this node's local knowledge
        const auto proposed_match = std::min(r.last_log_idx, state->last_idx);
```

Without this corresponding change in the TLA I was seeing trace validation failures for `fancy_election.2`, which I tracked down to a discrepancy in `matchIndex`. We try to match a `send_append_entries` step where the node says its `match_idx` is 12, yet the model has incremented it to 14. It was set to 14 when processing an AER ACK, despite the receiver's log only going as far as 12.

I thought this was an unnecessary safety check (as the comment says), but now it looks more worrying. An ACK extending past your own log means the receiver has a longer version of the previous term, which you are about to overwrite. They would overwrite it if you sent them an appropriate AE, but if you increment `match_idx` without that then you potentially advance commit unsafely.

Reply:
This is a receiver-side safety check that's doing only part of the necessary check. The above change is what sender should actually be doing. I'll remove this `min` from both, and see if anything breaks.